component/service/src/main/java/org/exoplatform/social/processor/EmbedProcessor.java

```
@@ -0,0 +1,50 @@
+package org.exoplatform.social.processor;
+
+import org.apache.commons.lang3.StringUtils;
+import org.exoplatform.container.xml.InitParams;
+import org.exoplatform.services.log.ExoLogger;
+import org.exoplatform.services.log.Log;
+import org.exoplatform.social.core.BaseActivityProcessorPlugin;
+import org.exoplatform.social.core.activity.model.ExoSocialActivity;
+import org.exoplatform.social.service.rest.LinkShare;
+
+
+public class EmbedProcessor extends BaseActivityProcessorPlugin {
+    private static final Log LOG = ExoLogger.getLogger(EmbedProcessor.class);
+
+    public EmbedProcessor(InitParams params) {
+        super(params);
+    }
+
+    public void processActivity(ExoSocialActivity activity) {
+        try {
+            int firstIndex = 0;
+            int lastIndex = 0;
+            String defaultValue = activity.getTitle();
+            String firstMessage = "";
+            String lastMessage = "";
+            String newValue = "";
+            while (firstIndex != -1) {
+                firstIndex = defaultValue.indexOf("<oembed>", firstIndex);
+                lastIndex = defaultValue.indexOf("</oembed>", lastIndex);
+                if (firstIndex != 0 && lastIndex !=0) {
+                    firstMessage = defaultValue.substring(0,firstIndex);
+                    lastMessage = defaultValue.substring(lastIndex + 9,defaultValue.length() );
+                    newValue = defaultValue.substring(firstIndex + 8, lastIndex);
+                }
+                firstIndex = -1;
+                lastIndex = -1;
+
+            }
+            String url = newValue.split("\"")[1];
+
+            if (StringUtils.isNoneBlank(url)) {
+                String html = LinkShare.getInstance(url).getMediaObject().getHtml();
+                String newTitleActivity = firstMessage + html +lastMessage;
+                activity.setTitle(newTitleActivity);
+            }
+        } catch (Exception e) {
+
```

Exception should never be swallowed. At least add an error log.